### PR TITLE
add reverse support

### PR DESCRIPTION
**Pull request by raimohanska**

### CODE REVIEW

**Comment by raimohanska** [comment]
Allows the bot to reverse and resume after hitting a wall

File: Assets/Scripts/CarController.cs [hunk 1]
```diff
@@ -99,6 +100,8 @@ private void Update()
     {
         UpdateWheelPoses();
         ProcessBotCommands();
+        var velocityAngle = Vector3.Angle(rigidBody.transform.forward, rigidBody.velocity);
+        velocity = rigidBody.velocity.magnitude * ((velocityAngle <= 90) ? 1 : -1);
```
Comment (mikkomultanen) [inline Assets/Scripts/CarController.cs]: Maybe this could be :
velocity = Vector3.Dot(rigidBody.transform.forward, rigidBody.velocity)

**Review by mikkomultanen — CHANGES_REQUESTED** [review]

File: Assets/Scripts/CarController.cs [hunk 2]
```diff
@@ -21,6 +21,7 @@ public class CarController : MonoBehaviour
     public float maxSteerAngle = 30;
     public float motorForce = 50;
     public float maxAngleChangePerSecond = 10;
+    public float velocity;
```
Comment (mikkomultanen) [inline Assets/Scripts/CarController.cs:25]: Should have [HideInInspector] annotation

**Review by mikkomultanen — CHANGES_REQUESTED** [review]

File: Assets/Scripts/CarController.cs (hunk 2 above)
Comment (raimohanska) [reply Assets/Scripts/CarController.cs:25]: Done!

File: Assets/Scripts/CarController.cs (hunk 1 above)
Comment (raimohanska) [reply Assets/Scripts/CarController.cs]: Done. Had totally forgot my vector math...

**Comment by raimohanska** [comment]
Thanks for reviewing! Really appreciated!

**Pull request merged**